docs/data/base/components/slider/slider.md

```diff
@@ -1,193 +1,243 @@
 
-The most basic Slider is _continuous_, which means it does not have pre-defined (_discrete_) values for the user to select from.
-This is suitable for situations in which an approximate value is good enough for the user, such as brightness or volume.
+```tsx
+<Slider.Root minStepsBetweenValues={2} step={5} defaultValue={[10, 20]}>
+  <Slider.Control>
+    <Slider.Track>
+      <Slider.Indicator />
+      <Slider.Thumb />
+      <Slider.Thumb />
+    <Slider.Track/>
+  </Slider.Control>
+</Slider.Root>
+```
 
-But if your users need more precise options, you can create a discrete Slider that snaps the thumb to pre-defined stops along the bar.
-Make sure to adjust the `shiftStep` prop (the granularity with which the slider can step when using Page Up/Down or Shift + Arrow Up/Down) to a value divadable with the `step`.
+## Vertical
 
-To generate a mark for each stop, use `marks={true}`:
+To create vertical sliders, set the `orientation` prop to `"vertical"`. This will track thumb movements vertically instead of horizontally.
 
-{{"demo": "DiscreteSlider.js"}}
+```jsx
+<Slider.Root orientation="vertical">{/* Subcomponents */}</Slider.Root>
+```
 
-#### Custom marks
+{{"demo": "VerticalSlider.js"}}
 
-You can create custom marks by providing a rich array to the `marks` prop:
+## RTL
 
-{{"demo": "DiscreteSliderMarks.js"}}
+Set the `direction` prop to `'rtl'` to change the slider's direction for right-to-left languages:
 
-#### Restricted values
+```jsx
+<Slider.Root direction="rtl">{/* Subcomponents */}</Slider.Root>
+```
 
-If the user should only be able to select from the values provided with the `marks` prop, add `step={null}` to disable all other options:
+In a RTL Slider, <kbd class="key">Left Arrow</kbd> increases the value while <kbd class="key">Right Arrow</kbd> decreases the value.
 
-{{"demo": "DiscreteSliderValues.js"}}
+{{"demo": "RtlSlider.js"}}
 
-### Range slider
+## Overriding default components
 
-To let users set the start and end of a range on a Slider, provide an array of values to the `value` or `defaultValue` prop:
+Use the `render` prop to override the rendered elements with your own components.
 
-{{"demo": "RangeSlider.js"}}
+```jsx
+<Slider.Control render={(props) => <MyCustomTrack {...props} />}> />
+```
 
-### Value label
+All subcomponents accept the `render` prop.
 
-A label for the value can be rendered around the thumb by using the optional `slots` prop with the `valueLabel` slot.
-These are typical use cases for showing the value label:
+The `Slider.Thumb` component's `render` prop contains an additional `inputProps` argument for rendering an `input` element attached to the thumb:
 
-- always
-- only when hovering over the thumb (using CSS)
-- while interacting with the thumb (hovering or dragging)
+```jsx
+<Slider.Thumb
+  render={(props, inputProps) => {
+    const { children, ...rest } = props;
```

Comment:
We might want to change the naming convention in the codebase for this, but for now since it's one of the only variations we have in the codebase, let's stick to it.

```suggestion
    const { children, ...other } = props;
```

Done in 89821a41b88ca47826bfbc0366f69abe40b4663c

(I found this from reading the docs in MUI X Charts initially, I then did a quick search to see the codebase patterns, and I found this in Base UI).